src/controllers/trust.details.controller.ts

```diff
@@ -54,34 +86,44 @@ const post = (
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
 
-    const trustId = req.params['id'];
-
     //  get trust data from session
     const appData: ApplicationData = getApplicationData(req.session);
-    const trusts: Trust[] = appData[TrustKey] ?? [];
-    const trustIndex = trusts.findIndex(trust => trust.trust_id === trustId);
-
-    // remove trust from the array of trusts   if it is present in the trusts array
-    if (trustIndex >= 0) {
-      trusts?.splice(trustIndex, 1);
-    }
 
     //  map form data to session trust data
-    const details = mapperToSessionService.mapDetailToSession(req.body);
-
-    //  add new details to session trusts array
+    const details = mapperDetails.mapDetailToSession(req.body);
     if (!details.trust_id) {
-      details.trust_id = String(trusts.length + 1);
+      details.trust_id = mapperDetails.generateTrustId(appData);
     }
 
+    //  remove trust for update from the array of trusts  if it is present in the trusts array
+    const trusts: Trust[] = (appData[TrustKey] ?? [])
+      .filter((trust: Trust) => trust.trust_id !== details.trust_id);
+
     trusts.push(details);
 
+    //  update trusts in beneficial owners
```

Comment:
Just a general point - this 'post' function is now getting quite long. If it was Java I'd suggest splitting up into a number of private methods, just to improve readability.

Reply:
agree move update app data functionality to separate functions